src/test/java/ru/itmo/sd/deadliner2bot/service/TaskFetchingServiceJpaTest.java

```diff
@@ -0,0 +1,178 @@
+
+    @BeforeEach
+    public void setUp() {
+        now = LocalDateTime.of(2000, 1, 1, 0, 0, 0);
+    }
+
+    @Test
+    @DisplayName(value = "Daily notifications in range fetched")
+    public void dailyNotificationInRangeFetched() {
+        Chat chat = createChat();
+        DailyNotification dailyNotification = createDailyNotification(chat, now.plusSeconds(storedRangeSec / 2));
+
+        Set<DailyNotification> dailyNotifications = taskFetchingService
+                .getDailyNotifications(now.plusSeconds(storedRangeSec));
+        assertThat(dailyNotifications).contains(dailyNotification);
+        Set<TodoNotification> todoNotifications = taskFetchingService
+                .getTodoNotifications(now.plusSeconds(storedRangeSec));
+        assertThat(todoNotifications).isEmpty();
+    }
+
+    @Test
+    @DisplayName(value = "Daily notifications out of range not fetched")
+    public void dailyNotificationOutOfRangeNotFetched() {
+        Chat chat = createChat();
+        createDailyNotification(chat, now.plusSeconds(2L * storedRangeSec));
+
+        Set<DailyNotification> dailyNotifications = taskFetchingService
+                .getDailyNotifications(now.plusSeconds(storedRangeSec));
+        assertThat(dailyNotifications).isEmpty();
+        Set<TodoNotification> todoNotifications = taskFetchingService
+                .getTodoNotifications(now.plusSeconds(storedRangeSec));
+        assertThat(todoNotifications).isEmpty();
+    }
+
+    @Test
+    @DisplayName(value = "TODO notifications in range fetched")
+    public void todoNotificationInRangeNotCompletedFetched() {
+        Chat chat = createChat();
+        TodoNotification todoNotification = createTodoWithNotification(chat, now.plusSeconds(storedRangeSec / 2));
+
+        Set<DailyNotification> dailyNotifications = taskFetchingService
+                .getDailyNotifications(now.plusSeconds(storedRangeSec));
+        assertThat(dailyNotifications).isEmpty();
+        Set<TodoNotification> todoNotifications = taskFetchingService
+                .getTodoNotifications(now.plusSeconds(storedRangeSec));
+        assertThat(todoNotifications).contains(todoNotification);
+    }
+
+    @Test
+    @DisplayName(value = "TODO notifications out of range not fetched")
+    public void todoNotificationNotInRangeNotFetched() {
+        Chat chat = createChat();
+        createTodoWithNotification(chat, now.plusSeconds(2L * storedRangeSec));
+
+        Set<DailyNotification> dailyNotifications = taskFetchingService
+                .getDailyNotifications(now.plusSeconds(storedRangeSec));
+        assertThat(dailyNotifications).isEmpty();
+        Set<TodoNotification> todoNotifications = taskFetchingService
+                .getTodoNotifications(now.plusSeconds(storedRangeSec));
+        assertThat(todoNotifications).isEmpty();
+    }
+
+    @Test
+    @DisplayName(value = "Daily notification added in scheduled thread pool")
+    public void dailyNotificationAddedInScheduledThreadPool() {
```

Comment:
Will be good to check, that notifications were removed after fetching.

Reply:
Good point.

Reply:
Added in the follow-up.